### PR TITLE
Enable FFTS builds for ARM (Xavier, QNX)

### PR DESCRIPTION
**Pull request by JanuszL**
- switches to FFTS fork with a fix for non x86 SSE capable machines
- turns off DISABLE_DYNAMIC_CODE for all but x86 builds
- enable FFTS to build for Xavier and QNX config

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It adds new feature needed because we want to build FFTS and audio for ARM as well

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     Enables FFTS build for ARM
     Turns off DISABLE_DYNAMIC_CODE for all but x86 builds
     Switches to FFTS fork with a fix for non x86 SSE capable machines
 - Affected modules and functionalities:
     CMake
 - Key points relevant for the review:
     NA
 - Validation and testing:
     CI
 - Documentation (including examples):
     NA

**JIRA TASK**: *[NA]*

### CODE REVIEW

File: cmake/Dependencies.common.cmake [hunk 1]
```diff
@@ -161,6 +161,10 @@ if (BUILD_FFTS)
   set(GENERATE_POSITION_INDEPENDENT_CODE ON CACHE BOOL "-fPIC")
   set(ENABLE_SHARED OFF CACHE BOOL "shared library target")
   set(ENABLE_STATIC ON CACHE BOOL "static library target")
+  # dynamic machine code generation works only for x86
+  if(ARCH)
```
Comment (mzient) [inline cmake/Dependencies.common.cmake]: Shouldn''t it be something more specific? like `if(ARCH_ARM)` or `if(ARCH MATCHES "ARM|AARCH")'?

Comment (klecki) [reply cmake/Dependencies.common.cmake]: Maybe use `CMAKE_SYSTEM_PROCESSOR` as in the platform/ cmake files.

Comment (klecki) [reply cmake/Dependencies.common.cmake]: And don't bother with defining `ARCH` in the scripts.

Comment (JanuszL) [reply cmake/Dependencies.common.cmake]: We may have aarch64-linux, or aarch64-qnx or anything else. I will follow @klecki suggestion.

File: docker/Dockerfile [hunk 2]
```diff
@@ -33,6 +33,8 @@ ENV CC=${CC}
 ENV CXX=${CXX}
 # Optional build arguments
```
Comment (klecki) [inline docker/Dockerfile]: Do we still need those `ARCH` things? I would remove them if no.

Comment (JanuszL) [reply docker/Dockerfile]: If our CMake accepts and uses this value I would like to have an ability to pass it in docker. Just in case.

Comment (klecki) [reply docker/Dockerfile]: But you removed them, now it's only created through our build scripts and the name is IMO too generic.

Comment (JanuszL) [reply docker/Dockerfile]: After discussion we will leave it as it is as it is used in other places as well.

**Pull request merged**